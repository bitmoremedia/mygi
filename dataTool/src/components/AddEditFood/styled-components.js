import styled from 'styled-components'

export const Container = styled.div`
  margin-bottom: 5px;
`

export const ButtonWrapper = styled.div`
  margin-top: 7px;
`

export const ErrorMsg = styled.div`
  color: red;
  margin-top: 5px;
`

export const Input = styled.input`
  margin-top: 5px;
  margin-bottom: 5px;
  outline: none;
  background-color: transparent;
  border: 0.1rem solid #d1d1d1;
  border-radius: 5px;
  box-shadow: none;
  box-sizing: inherit;
  height: 30px;
  padding-left: 10px;
  width: 100%;
`

export const Select = styled.select`

  background: url(data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" height="14"â€¦h fill="#d1d1d1" d="M9.37727 3.625l5.08154 6.93523L19.54036 3.625"/></svg>) center right no-repeat;
  padding-right: 3.0rem;

  margin-top: 5px;
  margin-bottom: 5px;
  outline: none;
  background-color: transparent;
  border: 0.1rem solid #d1d1d1;
  box-shadow: none;
  box-sizing: inherit;
  height: 30px;
  padding-left: 10px;
  width: 100%;

`
