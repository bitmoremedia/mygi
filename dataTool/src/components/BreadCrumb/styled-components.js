import styled from 'styled-components';

export const List = styled.ul`
  padding: 0px;
  margin-top: 15px;
  margin-bottom: 10px;
`;

export const ListItem = styled.li`
  display: inline;
  padding-right: 10px;
`;
