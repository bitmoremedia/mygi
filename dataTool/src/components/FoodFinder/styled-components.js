import styled from 'styled-components'

export const Container = styled.div`
  padding: 5px;
`

export const Heading = styled.h1`
`

export const SubHeading = styled.h4`
`
