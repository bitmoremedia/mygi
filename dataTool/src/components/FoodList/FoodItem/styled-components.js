import styled from 'styled-components'

export const Item = styled.div`
  padding: 5px;
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: space-between
`

export const FoodName = styled.div`
  white-space: nowrap;
`

export const EditButton = styled.div`
  margin-left: 10px;
`
