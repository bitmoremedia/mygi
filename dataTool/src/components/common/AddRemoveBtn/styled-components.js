import styled from 'styled-components'

export const AddButton = styled.span`
  color: green;
  cursor: pointer;
`

export const RemoveButton = styled.span`
  color: 'red';
  cursor: pointer;
`
