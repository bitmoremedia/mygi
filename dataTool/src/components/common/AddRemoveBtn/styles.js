import styled from 'styled-components';

export const Button2 = styled.button`
  background-color: ${props => props.type  === 'add' ? 'green' : 'red'};
  color: white;
  cursor: pointer;
  border-radius: 1px;
  border: 0px;
  padding: 0px;
  font-size: 12px;
  padding: 1px 3px 0px 3px;
  &:focus {
      outline:none;
  }
`;

export const Button = styled.span`
  color: ${props => props.type  === 'add' ? 'green' : 'red'};
  cursor: pointer;
`;
