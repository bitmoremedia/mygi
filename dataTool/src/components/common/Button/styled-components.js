import styled from 'styled-components'

export const Button = styled.button`
  background-color: black;
  color: white;
  cursor: pointer;
  border-radius: 1px;
  border: 0px;
  padding: 5px 10px 5px 10px;
  font-size: 14px;
  &:focus {
      outline:none;
  }
`

export const DangerButton = styled(Button)`
  background-color: red;
`
