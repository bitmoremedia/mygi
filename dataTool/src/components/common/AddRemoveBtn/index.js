import React, { PropTypes } from 'react'

import { AddButton, RemoveButton } from './styled-components'

const AddRemoveBtn = ({ onClick, type }) => {
  if (type === 'add') {
    return (
      <AddButton onClick={onClick}>
        Assign
      </AddButton>
    )
  }
  return (
    <RemoveButton onClick={onClick}>
      Un-Assign
    </RemoveButton>
  )
}

AddRemoveBtn.propTypes = {
  onClick: PropTypes.func,
  type: PropTypes.oneOf(['add', 'remove']),
}

AddRemoveBtn.defaultProps = {
  onClick: undefined,
  type: 'add',
}

export default AddRemoveBtn
