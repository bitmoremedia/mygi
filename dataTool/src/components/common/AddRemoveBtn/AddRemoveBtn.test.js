import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import AddRemoveBtn from './index'
import { AddButton } from './styled-components'

test('component renders without error', () => {
  const wrapper = mount(
    <AddRemoveBtn />,
  )
  expect(wrapper.exists()).toBe(true)
})

// INTERACTIVITY TESTS

test('onclick function is fired', () => {
  const onClick = jest.fn()
  const wrapper = mount(
    <AddRemoveBtn onClick={onClick} />,
  )
  expect(onClick).not.toHaveBeenCalled()
  wrapper.find(AddButton).simulate('click')
  expect(onClick).toHaveBeenCalled()
})

// SNAPSHOT TESTS

test('render add mode', () => {
  const component = renderer.create(
    <AddRemoveBtn type="add" />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('render remove mode', () => {
  const component = renderer.create(
    <AddRemoveBtn type="remove" />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
