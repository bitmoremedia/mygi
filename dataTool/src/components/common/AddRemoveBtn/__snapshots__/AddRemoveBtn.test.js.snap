exports[`test render add mode 1`] = `
<span
  className="bBeLnM"
  onClick={undefined}>
  Assign
</span>
`;

exports[`test render remove mode 1`] = `
<span
  className="gZIOPq"
  onClick={undefined}>
  Un-Assign
</span>
`;
